### PR TITLE
12.0 mig account loan

### PR DESCRIPTION
**Pull request by amcor**
cc @guadaltech

### CODE REVIEW

File: account_loan/model/account_loan.py [hunk 1]
```diff
@@ -0,0 +1,488 @@
...
+        self.journal_id = self.env['account.journal'].search([
+            ('company_id', '=', self.company_id.id),
+            ('type', '=', 'purchase' if self.is_leasing else 'general')
+        ], limit=1)
+        self.residual_amount = 0.0
+
+    @api.onchange('company_id')
+    def _onchange_company(self):
+        self._onchange_is_leasing()
+        self.interest_expenses_account_id = False
```
Comment (nikul-serpentcs) [inline account_loan/model/account_loan.py]: @amcor `self.interest_expenses_account_id = self.short_term_loan_account_id = self.long_term_loan_account_id = False`

**Review by nikul-serpentcs — COMMENTED** [review]
Minor change

File: account_loan/wizard/account_loan_pay_amount.py [hunk 2]
```diff
@@ -1,4 +1,3 @@
-# Copyright 2018 Creu Blanca
```
Comment (MiquelRForgeFlow) [inline account_loan/wizard/account_loan_pay_amount.py]: why?

**Review by MiquelRForgeFlow — COMMENTED** [review]
runbot is red. Will do you work to make it green?

File: account_loan/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# Copyright 2018 Creu Blanca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Account Loan management",
```
Comment (MiquelRForgeFlow) [inline account_loan/__manifest__.py:4]:
```suggestion
    "name": "Account Loan",
```

File: account_loan/readme/USAGE.rst [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+On a posted loan you can:
+
+* Create moves or invoices (according to the configuration)
+* Modify rates when needed (only unposted lines will be modified)
+* Reduce or cancel the debt of a loan / lease
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/92/11.0
```
Comment (MiquelRForgeFlow) [inline account_loan/readme/USAGE.rst]:
```suggestion
   :target: https://runbot.odoo-community.org/runbot/92/12.0
```

File: account_loan/wizard/account_loan_pay_amount.py (hunk 2 above)
Comment (amcor) [reply account_loan/wizard/account_loan_pay_amount.py]: Upps, I've no idea, a mistake removing encodings ? I revert it.

**Review by NuriaXifre — APPROVED** [review]
Functional review done! The module works as expected.

**Review by etobella — APPROVED** [review]
LGTM :+1: Could you add #847 commits?

**Review by jarroyomorales — APPROVED** [review]

**Comment by jarroyomorales** [comment]
@OCA/accounting-maintainers Can this be merged?

**Comment by marylla** [comment]
@pedrobaeza Hi Pedro, is it possible to merge this soon? Thanks a lot!

**Comment by marylla** [comment]
@pedrobaeza Do you know what happened with this merge attempt?

**Comment by pedrobaeza** [comment]
It failed. You have to rebase and see what's the problem.

**Comment by amcor** [comment]
Hi @pedrobaeza I have done the rebase but the coverage .... I dont know if i have done something wrong ..

**Comment by pedrobaeza** [comment]
Please never do merge operations in a pull request. This messes up commit history. Codecov failure is because the module doesn't have enough tests, but this is not mandatory.

**Review by pedrobaeza — DISMISSED** [review]
Remove merge commit doing a proper rebase

**Comment by amcor** [comment]
sorry @pedrobaeza , and now ?

**Comment by marylla** [comment]
@pedrobaeza @amcor It is unfortunately still not merged. What happened? :( Sorry for the recurring questions about this.

**Comment by marylla** [comment]
@amcor maybe rebase is needed again?

Now, I found out how to see the failing reasons in travis log. 

<pre>
Traceback (most recent call last):
`   File "/home/travis/build/OCA/account-financial-tools/account_payment_netting/tests/test_account_payment_netting.py", line 131, in test_1_payment_netting_neutral
`     invoices.action_invoice_open()
`   File "/home/travis/odoo-12.0/addons/account/models/account_invoice.py", line 990, in action_invoice_open
`     to_open_invoices.action_move_create()
`   File "/home/travis/build/OCA/account-financial-tools/account_loan/model/account_invoice.py", line 24, in action_move_create
`     if self.loan_line_id:
`   File "/home/travis/odoo-12.0/odoo/fields.py", line 1003, in __get__
`     record.ensure_one()
`   File "/home/travis/odoo-12.0/odoo/models.py", line 4744, in ensure_one
`     raise ValueError("Expected singleton: %s" % self)
` ValueError: Expected singleton: account.invoice(30, 31, 32, 33)
ERROR: test_2_payment_netting_inbound (odoo.addons.account_payment_netting.tests.test_account_payment_netting.TestAccountNetting)
` Test AR > AP
Traceback (most recent call last):
`   File "/home/travis/odoo-12.0/odoo/models.py", line 4741, in ensure_one
`     _id, = self._ids
` ValueError: too many values to unpack (expected 1)
</pre>

**Comment by marylla** [comment]
@pedrobaeza Hi Pedro, can we try the merge again? The coverage test fails but as I understand from your comment https://github.com/OCA/account-financial-tools/pull/833#issuecomment-602669718 it is not mandatory for a migration?! Maybe it works this time and I don't have to bother you again about this one. ;)

**Pull request merged**